### PR TITLE
Add support for custom web / api staging urls for eigen

### PR DESCRIPTION
**Pull request by orta**
Makes it easier for non-cocoa devs to move to test Eigen support without deploying to staging or production:

screen shot 2016-02-23 at 11 25 03 am

screen shot 2016-02-23 at 11 24 34 am

screen shot 2016-02-23 at 11 25 06 am

### CODE REVIEW

**Comment by orta** [comment]
got some legit build fails local

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%">1 Error</th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:no_entry_sign:</td>
      <td><p>No CHANGELOG changes made</p>
</td>
    </tr>
    
  </tbody>
</table>
  

  
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%">2 Warnings</th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td><p>Needs testing on a Phone if change is non-trivial</p>
</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td><h4>Found unstubbbed networking requests</h4>

<ul>
<li><a href="https://github.com/search?q=ARArtworkViewControllerBuyButton+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARArtworkViewControllerBuyButtonSpec</a> in <code>test_buy_button__posts_order_if_artwork_has_no_edition_sets</code></li>
<li><a href="https://github.com/search?q=ARCreateAccountViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARCreateAccountViewControllerSpec</a> in <code>test_warning_view__displays_a_message</code></li>
<li><a href="https://github.com/search?q=ARCreateAccountViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARCreateAccountViewControllerSpec</a> in <code>test_warning_view__displays_a_message</code></li>
<li><a href="https://github.com/search?q=AREmbeddedModelPreviewViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">AREmbeddedModelPreviewViewControllerSpec</a> in <code>test_iPhone_peek__shows_favorite_and_follow_buttons_for_logged_in_user</code></li>
<li><a href="https://github.com/search?q=AREmbeddedModelPreviewViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">AREmbeddedModelPreviewViewControllerSpec</a> in <code>test_iPhone_peek__shows_favorite_and_follow_buttons_for_logged_in_user</code></li>
<li><a href="https://github.com/search?q=AREmbeddedModelPreviewViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">AREmbeddedModelPreviewViewControllerSpec</a> in <code>test_iPhone_peek__passes_artwork_to_share_controller</code></li>
<li><a href="https://github.com/search?q=ARFairViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARFairViewControllerSpec</a> in <code>test_creates_an_NSUserActivity</code></li>
<li><a href="https://github.com/search?q=ARGeneViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARGeneViewControllerSpec</a> in <code>test_creates_an_NSUserActivity</code></li>
<li><a href="https://github.com/search?q=ARGeneViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARGeneViewControllerSpec</a> in <code>test_creates_an_NSUserActivity</code></li>
<li><a href="https://github.com/search?q=ARTopMenuViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARTopMenuViewControllerSpec</a> in <code>test_navigation__favorites__logged_in__tab_behavior__already_selected__pops_to_root</code></li>
</ul>
</td>
    </tr>
    
  </tbody>
</table>
  

  


<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/KrauseFx/danger/">danger</a>
</p>

**Comment by orta** [comment]
This fail is just danger - it's actually green - it would be fixed by master danger - but rugged makes that not working ATM.

**Comment by ashfurrow** [comment]
Looks good 👍 I imagine it'd be nice to set more than one, _then_ restart, but that's trickier (I know since I often have to change three things on eidolon's admin, but it is like a _super_ nice-to-have).

**Comment by orta** [comment]
That's going to let people shoot themselves in the foot too easily I'm afraid

File: Artsy_Tests/Networking_Tests/API_Modules/ARUserManagerTests.m [hunk 1]
```diff
@@ -161,7 +165,7 @@ + (void)clearUserData:(ARUserManager *)manager useStaging:(id)useStaging;
         });
 
         it(@"explicitly sets staging default to yes", ^{
-            expect([[NSUserDefaults standardUserDefaults] valueForKey:ARUseStagingDefault]).to.beNil();
+            expect([[NSUserDefaults standardUserDefaults] objectForKey:ARUseStagingDefault]).to.beTruthy();
```
Comment (orta) [inline Artsy_Tests/Networking_Tests/API_Modules/ARUserManagerTests.m]: this is caused by us now running `ARDefaults setup` on test launch now

File: Artsy/Networking/ARRouter.m [hunk 2]
```diff
@@ -28,12 +28,26 @@
...
+
 
 @implementation ARRouter
 
 + (void)setup
 {
-    artsyHosts = [NSSet setWithObjects:@"art.sy", @"artsyapi.com", @"artsy.net", @"m.artsy.net", @"staging.artsy.net", @"m-staging.artsy.net", nil];
+    NSString *productionHost = hostFromString(ARBaseDesktopWebURL);
+    NSString *productionMobile = hostFromString(ARBaseMobileWebURL);
+    NSString *productionAPI = hostFromString(ARBaseMobileWebURL);
```
Comment (alloy) [inline Artsy/Networking/ARRouter.m]: Maybe these constants should just really be NSURL instances?

File: Artsy_Tests/Networking_Tests/API_Modules/ARUserManagerTests.m [hunk 3]
```diff
@@ -10,7 +10,11 @@ + (void)clearUserData;
 + (void)clearUserData:(ARUserManager *)manager useStaging:(id)useStaging;
 @end
 
-SpecBegin(ARUserManager);
+// This makes a lot of global changes to the  NSUserDefaults shared
+// user defaults, and so is ran at the end.
+// FIXME: Migrate to using DI for defaults.
+
+SpecBegin(ZARUserManager);
```
Comment (alloy) [inline Artsy_Tests/Networking_Tests/API_Modules/ARUserManagerTests.m]: Hah, low tech

File: Artsy/Networking/ARRouter.m (hunk 2 above)
Comment (orta) [inline Artsy/Networking/ARRouter.m]: I'm pretty nonchalant on that, I mean, we store them as `const` - wouldn't we have o move them to `+initialize` then? not sure ewe'd gain much for the extra code

Comment (alloy) [inline Artsy/Networking/ARRouter.m]: Yup. It’s mainly the ‘URL’ suffix that makes me itch :)

Comment (orta) [inline Artsy/Networking/ARRouter.m]: me too buddy, me too

**Pull request merged**